### ISSUE TITLE
"Activating Extensions..." when trying to compile/preview pdf

### ISSUE THREAD

**Issue by Jazzpirate** [James-Yu/LaTeX-Workshop#3967]
> _Please fill the following fields with a star (*) and provide as much related information as possible._

## Pre-checks*
> _Please change the following `[ ]` to `[x]` for confirmation._
- [x] The issue has not been reported in this repository.
- [x] The issue remains after disabling all other extensions and restarting Visual Studio Code.
- [x] The [FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) cannot address the issue.
- [x] The issue is not related to compiling a document, or the document can be successfully compiled in the OS terminal but not in Visual Studio Code with this extension.

## Environment*
> _Please write exact version numbers instead of descriptors such as `latest`._
- Operating System: Ubuntu 22.04.2 LTS
- Visual Studio Code Version: 1.80.1
- LaTeX Workshop Version: 9.13.0
- TeX Distribution Version: TeX Live 2023

> _Please list the environment and version number if you are using VSCodium, Snap or Flatpack versions of Visual Studio Code, and/or Visual Studio Code Remote Containers/SSH/WSL._
- I am using XXX whose version is YY.ZZ

## The Issue*
> _Please briefly describe the issue you come across._
- Extension seems to fail to activate. The "LaTeX Workshop" output channel is empty. Static syntax highlighting works, but on pressing the "Build LaTeX Project" or "View LaTeX PDF File" buttons, all that happens is that the VSCode status bar shows "Activating Extensions..." for a few seconds.
- Reproduced on a colleagues macbook as well, so likely unrelated to the OS (but also using TeXLive 2023). The issue only appeared over the last few days, so likely related to the latest version
- I saw your commit regarding the port number. I set the corresponding setting to the number `0` rather than the string `"0"`, but that didn't fix the problem either.

### Reproduction Steps
> _Please list out the steps to reproduce your bug. Include relevant environmental variables or any other configuration._
1. Open any tex file and press one of the above buttons

### Expected Behavior
> _What were you expecting to see? Include any relevant examples or documentation links._
- Actual compilation / PDF preview

### LaTeX Workshop Output*
> _Please paste the whole log messages below, not parts of ones. The log should start with `New log placeholder %WS1% registered`._
```
(None)
```

### Developer Tools Console
> _Please paste the whole log messages below, not parts of ones. This console logs can sometimes be very important in many cases. To access the log, click `Help` -> `Toggle Developer Tools` -> `Console`._
```
TMScopeRegistry.ts:46 Overwriting grammar scope name to file mapping for scope text.tex.
Old grammar file: file:///snap/code/134/usr/share/code/resources/app/extensions/latex/syntaxes/TeX.tmLanguage.json.
New grammar file: file:///home/jazzpirate/.vscode/extensions/james-yu.latex-workshop-9.13.0/syntax/TeX.tmLanguage.json
TMScopeRegistry.ts:46 Overwriting grammar scope name to file mapping for scope text.tex.latex.
Old grammar file: file:///snap/code/134/usr/share/code/resources/app/extensions/latex/syntaxes/LaTeX.tmLanguage.json.
New grammar file: file:///home/jazzpirate/.vscode/extensions/james-yu.latex-workshop-9.13.0/syntax/LaTeX.tmLanguage.json
TMScopeRegistry.ts:46 Overwriting grammar scope name to file mapping for scope text.bibtex.
Old grammar file: file:///snap/code/134/usr/share/code/resources/app/extensions/latex/syntaxes/Bibtex.tmLanguage.json.
New grammar file: file:///home/jazzpirate/.vscode/extensions/james-yu.latex-workshop-9.13.0/syntax/Bibtex.tmLanguage.json
TMScopeRegistry.ts:46 Overwriting grammar scope name to file mapping for scope text.tex.markdown_latex_combined.
Old grammar file: file:///snap/code/134/usr/share/code/resources/app/extensions/latex/syntaxes/markdown-latex-combined.tmLanguage.json.
New grammar file: file:///home/jazzpirate/.vscode/extensions/james-yu.latex-workshop-9.13.0/syntax/markdown-latex-combined.tmLanguage.json
TMScopeRegistry.ts:46 Overwriting grammar scope name to file mapping for scope source.cpp.embedded.latex.
Old grammar file: file:///snap/code/134/usr/share/code/resources/app/extensions/latex/syntaxes/cpp-grammar-bailout.tmLanguage.json.
New grammar file: file:///home/jazzpirate/.vscode/extensions/james-yu.latex-workshop-9.13.0/syntax/cpp-grammar-bailout.tmLanguage.json
log.ts:421  INFO [perf] Render performance baseline is 71ms
log.ts:441   ERR Error: Aborted onWillSaveTextDocument-event after 1750ms
    at mainThreadSaveParticipant.ts:37:18
```

## Anything Else?
> _Add any other context about the problem below._
-

**Comment by Jazzpirate** [comment]
I can reproduce it with the workspace being a directory `test` with a single file `test.tex` containing only the following:
```
\documentclass{article}

\begin{document}
foo
\end{document}
```
(and all extensions other than LaTeX Workshop deactivated)

**Comment by Jazzpirate** [comment]
I have a suspicion. When I switched back to VSCode after going for lunch, the log output wasn't empty anymore. After restarting VSCode I'm back to the same problem. So it *might* be, that it doesn't fail to load, it just takes extremely long to do so.

I noticed your latest version now uses kpsewhich for various things that it didn't before. Important question: *In which PWD* is the kpsewhich process called? My TEXINPUTS tend to include `.//` (i.e. all subdirectories of the PWD), which means if kpsewhich is called e.g. in `~` or `/`, it can take an eternity to return results....
